security-framework/src/secure_transport.rs

```diff
@@ -281,6 +291,10 @@ impl<S> MidHandshakeClientBuilder<S> {
                         let err = Error::from_code(errSecTrustSettingDeny);
                         return Err(ClientHandshakeError::Failure(err));
                     }
+                    TrustResult::RecoverableTrustFailure if danger_accept_invalid_certs => {
```

Comment:
Actually, it seems like we arguably would want to skip the trust check entirely if we have danger_accept_invalid_certs set, right?

Reply:
Honestly, I'm not really sure. According to the [apple docs](https://developer.apple.com/documentation/security/sectrustresulttype) the `RecoverableTrustFailure` is the only one that we (as a piece of software) should be able to make a judgement call on. But that might be more directed at a high level app, and less at a low-level lib. If I explicitly decided to [deny](https://developer.apple.com/documentation/security/sectrustresulttype/1400766-deny) a cert chain, should this lib respect that?

Given that this will probably be used from `rust-native-tls`, I doubt we can expose these nuances meaningfully. Furthermore, I think the linux/windows implementations from my PR there will blanket accept whatever.

I think I agree with you then, that we should skip the check entirely.